### PR TITLE
Implement skipTransaction feature

### PR DESCRIPTION
**Pull request by stwiname**
# Description
Add support for `skipTransaction` feature which allows skipping fetching transactions, this reduces network bandwidth as well as memory usage. https://github.com/subquery/documentation/pull/400

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## TODO
- Rename flag from `skipBlock` to `skipTransaction`
- Add yargs option

## Checklist

- [x] I have tested locally
- [x] I have performed a self review of my changes
- [x] Updated any relevant documentation
- [x] Linked to any relevant issues
- [x] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules https://github.com/subquery/subql/pull/2044
- [x] My code is up to date with the base branch
- [ ] I have updated relevant changelogs. [We suggest using chan](https://github.com/geut/chan/tree/main/packages/chan)

### CODE REVIEW

File: packages/node/src/ethereum/api.ethereum.ts [hunk 1]
```diff
@@ -260,38 +268,56 @@ export class EthereumApi implements ApiWrapper<EthereumBlockWrapper> {
...
+  }
+
+  async fetchBlocks(bufferBlocks: number[]): Promise<EthereumBlock[]> {
+    return Promise.all(bufferBlocks.map(async (num) => this.fetchBlock(num)));
+  }
+
+  async fetchBlocksLight(
+    bufferBlocks: number[],
+  ): Promise<LightEthereumBlock[]> {
+    console.log('FETCH BLOCKS LIGHT');
```
Comment (ianhe8x) [inline packages/node/src/ethereum/api.ethereum.ts:318]: remove debugs

File: packages/types/src/ethereum/interfaces.ts [hunk 2]
```diff
@@ -118,10 +116,14 @@ export type EthereumLog<T extends EthereumResult = EthereumResult> = {
...
 export type FlareLog<T extends FlareResult = FlareResult> = EthereumLog<T>;
 
-export type EthereumBlockWrapper = BlockWrapper<
-  EthereumBlock,
-  EthereumTransaction,
-  EthereumLog,
-  EthereumTransactionFilter,
-  EthereumLogFilter
->;
+export type LightEthereumLog<T extends EthereumResult = EthereumResult> = Omit<
```
Comment (ianhe8x) [inline packages/types/src/ethereum/interfaces.ts:119]: I don't know, does `EthereumLogLite` sound better?

File: packages/node/src/indexer/types.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
 // SPDX-License-Identifier: GPL-3.0
 
-import { Entity } from '@subql/types-ethereum';
+import {
+  Entity,
+  EthereumBlock,
+  LightEthereumBlock,
+} from '@subql/types-ethereum';
 
 export enum OperationType {
```
Comment (jiqiang90) [inline packages/node/src/indexer/types.ts:10]: I think we no longer need these types, but maybe also could remove in branch `sync-20230829-eth`

**Review by jiqiang90 — APPROVED** [review]

File: packages/types/src/ethereum/interfaces.ts (hunk 2 above)
Comment (stwiname) [reply packages/types/src/ethereum/interfaces.ts:151]: To me `LightEthereumLog` makes more sense, it follows the naming for substrate too

**Review by jiqiang90 — APPROVED** [review]

**Pull request merged**